master/buildbot/util/git.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+from distutils.version import LooseVersion
+
+
+class GitFeaturesMixin(object):
+
+    def setupGitFeatures(self):
+        self.gitInstalled = False
+        self.supportsBranch = False
+        self.supportsSubmoduleForce = False
+        self.supportsSubmoduleCheckout = False
+        self.supportsSshPrivateKeyAsEnvOption = False
+        self.supportsSshPrivateKeyAsConfigOption = False
+
+    def parseGitFeatures(self, version_stdout):
+
+        if 'git' not in version_stdout:
+            return
+
+        try:
+            version = version_stdout.strip().split(' ')[2]
+        except IndexError:
+            return
+
+        self.gitInstalled = True
+        if LooseVersion(version) >= LooseVersion("1.6.5"):
+            self.supportsBranch = True
+        if LooseVersion(version) >= LooseVersion("1.7.6"):
+            self.supportsSubmoduleForce = True
+        if LooseVersion(version) >= LooseVersion("1.7.8"):
+            self.supportsSubmoduleCheckout = True
+        if LooseVersion(version) >= LooseVersion("2.3.0"):
+            self.supportsSshPrivateKeyAsEnvOption = True
+        if LooseVersion(version) >= LooseVersion("2.10.0"):
+            self.supportsSshPrivateKeyAsConfigOption = True
+
+
+def adjustCommandParamsForSshPrivateKey(command, env, gitFeatures,
+                                        keyPath, sshWrapperPath=None):
+    if gitFeatures.supportsSshPrivateKeyAsConfigOption:
```

Comment:
you could just use self.supportsSshPrivateKeyAsConfigOption if you were inside the mixin

Reply:
Right. I was thinking about the mixin as strictly containing functionality related to supported feature mask. Perhaps the class should be renamed to GitMixin if it contains additional functions.

Reply:
right

Reply:
Should be good now.